modules/mapper-extras/src/test/resources/rest-api-spec/test/search-as-you-type/20_highlighting.yml

```diff
@@ -0,0 +1,202 @@
+setup:
+  - skip:
+      version: " - 7.1.0"
+      reason: "added in 7.1.0"
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          settings:
+            number_of_replicas: 0
+          mappings:
+            properties:
+              a_field:
+                type: search_as_you_type
+                analyzer: simple
+                max_shingle_size: 4
+              text_field:
+                type: text
+                analyzer: simple
+
+  - do:
+      index:
+        index: test
+        type: _doc
+        id: 1
+        body:
+          a_field: "quick brown fox jump lazy dog"
+          text_field: "quick brown fox jump lazy dog"
+
+  - do:
+      indices.refresh: {}
+
+---
+"phrase query":
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          query:
+            match_phrase:
+              a_field: "brown"
+          highlight:
+            fields:
+              a_field:
+                type: unified
+
+  - match: { hits.total: 1 }
+  - match: { hits.hits.0._source.a_field: "quick brown fox jump lazy dog" }
+  - match: { hits.hits.0._source.text_field: "quick brown fox jump lazy dog" }
+  - match: { hits.hits.0.highlight.a_field.0: "quick <em>brown</em> fox jump lazy dog" }
+
+---
+"bool prefix query":
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          query:
+            match_bool_prefix:
+              a_field: "brown fo"
+          highlight:
+            fields:
+              a_field:
+                type: unified
+
+  - match: { hits.total: 1 }
+  - match: { hits.hits.0._source.a_field: "quick brown fox jump lazy dog" }
+  - match: { hits.hits.0._source.text_field: "quick brown fox jump lazy dog" }
+  - match: { hits.hits.0.highlight.a_field.0: "quick <em>brown</em> fox jump lazy dog" }
```

Comment:
@andyb-elastic Is it unreasonable to expect `quick <em>brown</em> <em>fo</em>x jump lazy dog` highlight here?

Reply:
I know it's not nice to comment on already merged PR, that's why I opened this issue:
https://github.com/elastic/elasticsearch/issues/53744